alchemy/src/cloudflare/worker/mixed-mode.ts

```diff
@@ -0,0 +1,216 @@
+    external: ["cloudflare:*"],
+    write: false,
+  });
+  if (!result.outputFiles?.[0]) {
+    throw new Error("Failed to bundle worker.ts");
+  }
+  return new File([result.outputFiles[0].text], "worker.js", {
+    type: "application/javascript+module",
+  });
+}
+
+export async function createMixedModeProxy(input: {
+  name: string;
+  bindings: WorkerBindingSpec[];
+}) {
+  const api = await createCloudflareApi();
+  const script = await bundleWorkerScript();
+  const [token, subdomain] = await Promise.all([
+    createWorkersPreviewToken(api, {
+      name: input.name,
+      metadata: {
+        main_module: script.name,
+        compatibility_date: "2025-06-16",
+        bindings: input.bindings,
+        observability: {
+          enabled: false,
+        },
+      },
+      files: [script],
+      session: {
+        workers_dev: true,
+        minimal_mode: true,
+      },
+    }),
+    getAccountSubdomain(api),
+  ]);
+  return new MixedModeProxy(
+    `https://${input.name}.${subdomain}.workers.dev`,
+    token,
+    input.bindings,
+  );
+}
+
+export class MixedModeProxy {
+  server: Bun.Server;
+  constructor(
+    readonly url: string,
+    readonly token: string,
+    readonly bindings: WorkerBindingSpec[],
+  ) {
+    this.server = Bun.serve({
+      fetch: this.fetch,
+    });
+  }
+
+  get connectionString() {
+    return new URL(
+      `http://${this.server.hostname}:${this.server.port}`,
+    ) as MixedModeConnectionString;
+  }
+
+  fetch = async (req: Request) => {
+    const origin = new URL(req.url);
+    const url = new URL(origin.pathname, this.url);
+    url.search = origin.search;
+    url.hash = origin.hash;
+
+    const headers = new Headers(req.headers);
+    headers.set("cf-workers-preview-token", this.token);
+    headers.set("host", new URL(this.url).hostname);
+    headers.delete("cf-connecting-ip");
+
+    return await fetch(url, {
+      method: req.method,
+      headers,
+      body: req.body,
+      redirect: "manual",
+    });
+  };
+}
+
+async function createWorkersPreviewToken(
+  api: CloudflareApi,
+  input: {
+    name: string;
+    metadata: WorkerMetadata;
+    files: File[];
+    session: WranglerSessionConfig;
+  },
+) {
+  const session = await createWorkersPreviewSession(api);
+  const formData = new FormData();
+  formData.append("metadata", JSON.stringify(input.metadata));
+  for (const file of input.files) {
+    formData.append(file.name, file);
+  }
+  formData.append("wrangler-session-config", JSON.stringify(input.session));
+  const res = await api
+    .post(
+      `/accounts/${api.accountId}/workers/scripts/${input.name}/edge-preview`,
+      formData,
+      {
+        headers: {
+          "cf-preview-upload-config-token": session.token,
+        },
+      },
+    )
+    .then((res) =>
+      parseCloudflareResponse<{ preview_token: string }>(
+        res,
+        "Failed to create workers preview token",
+      ),
+    );
+  prewarm(session.prewarm, res.preview_token);
```

Comment:
Await?

Reply:
The way CF does it is "fire and forget" ([source](https://github.com/cloudflare/workers-sdk/blob/6c6afbd6072b96e78e67d3a863ed849c6aa49472/packages/wrangler/src/dev/create-worker-preview.ts#L338)) so figured we'd do the same